src/MongoDB.Driver.Core/MongoConnectionException.cs

```diff
@@ -99,6 +98,29 @@ public bool ContainsSocketTimeoutException
             }
         }
 
+        /// <summary>
+        /// Whether or not this exception contains a timeout exception.
+        /// </summary>
+        public bool ContainsTimeoutException
```

Comment:
I think we can mark `ContainsSocketTimeoutException` as obsolete

Reply:
I don't see why there should be two methods. How about just refactoring the one method to check for both kinds of timeout exceptions?

    public bool ContainsTimeoutException
    {
        get
        {
            for (var exception = InnerException; exception != null; exception = exception.InnerException)
            {
                if (exception is SocketException socketException &&
                  socketException.SocketErrorCode == SocketError.TimedOut)
                {
                    return true;
                }

                if (exception is TimeoutException)
                {
                    return true;
                }
            }
            return false;
        }
    }

Reply:
@MikalaiMazurenka done